### PR TITLE
Sscs 7809 resintatement judge

### PR DESCRIPTION
**Pull request by danhamilton**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/IntegrationTests/resources/config/application_it.properties [hunk 1]
```diff
@@ -21,3 +21,5 @@ evidence.welsh.hmctsImgVal="[userImage:welshhmcts.png]"
 
 cover.english=SSCS-cover-page.docx
 cover.welsh=TB-SCS-LET-WEL-00486.docx
+
+reinstatement_requests_feature_flag=false
```
Comment (srochani) [inline src/IntegrationTests/resources/config/application_it.properties:25]: Why this feature flag is disable, are we not including integration test for this reinstatement

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java [hunk 2]
```diff
@@ -124,11 +131,100 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
 
-    private void updateCaseAfterExtensionRefused(SscsCaseData caseData, String interlocReviewState, State state) {
+    @NotNull
+    private SscsCaseData updateCaseForDirectionType(CaseDetails<SscsCaseData> caseDetails, SscsCaseData caseData) {
+
+        System.out.println("HERE: " + caseData.getDirectionTypeDl().getValue().getCode());
+
+        if (DirectionType.PROVIDE_INFORMATION.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())) {
+
+            System.out.println("HEREA");
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java:141]: Please remove this System.out.println, use sl4j logs instead

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java [hunk 3]
```diff
@@ -124,11 +131,100 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
+            caseData.setInterlocReviewState(AWAITING_INFORMATION.getId());
+
+        } else if (getPreValidStates().contains(caseDetails.getState())
+                && DirectionType.APPEAL_TO_PROCEED.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())) {
+            System.out.println("HEREB");
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java:146]: same here

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java [hunk 4]
```diff
@@ -150,6 +151,9 @@ private void buildSscsDocumentFromScan(SscsCaseData sscsCaseData, State caseStat
 
                     if (!equalsIgnoreCase(scannedDocument.getValue().getType(), COVERSHEET)) {
                         SscsDocument sscsDocument = buildSscsDocument(sscsCaseData, scannedDocument, caseState);
+                        if (REINSTATEMENT_REQUEST.getValue().equals(sscsDocument.getValue().getDocumentType())) {
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java:154]: No feature flag check here

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java [hunk 5]
```diff
@@ -124,11 +131,100 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
+            caseData = updateCaseAfterExtensionRefused(caseData, AWAITING_ADMIN_ACTION.getId(), State.RESPONSE_RECEIVED);
+
+        } else if (DirectionType.REFUSE_EXTENSION.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())
+            && ExtensionNextEvent.SEND_TO_VALID_APPEAL.toString().equals(caseData.getExtensionNextEventDl().getValue().getCode())) {
+            System.out.println("HERED");
+            caseData = updateCaseAfterExtensionRefused(caseData, null, State.WITH_DWP);
+
+        } else if (DirectionTypeItemList.GRANT_REINSTATEMENT.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())
+            && reinstatementFeatureFlag) {
+            System.out.println("HERE GRANT");
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java:162]: Remove println

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java [hunk 6]
```diff
@@ -124,11 +131,100 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
+            caseData.setInterlocReviewState(AWAITING_ADMIN_ACTION.getId());
+
+        } else if (DirectionType.REFUSE_EXTENSION.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())
+            && ExtensionNextEvent.SEND_TO_LISTING.toString().equals(caseData.getExtensionNextEventDl().getValue().getCode())) {
+            System.out.println("HEREC");
+            caseData = updateCaseAfterExtensionRefused(caseData, AWAITING_ADMIN_ACTION.getId(), State.RESPONSE_RECEIVED);
+
+        } else if (DirectionType.REFUSE_EXTENSION.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())
+            && ExtensionNextEvent.SEND_TO_VALID_APPEAL.toString().equals(caseData.getExtensionNextEventDl().getValue().getCode())) {
+            System.out.println("HERED");
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java:157]: remove println

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java [hunk 7]
```diff
@@ -124,11 +131,100 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
+
+        } else if (getPreValidStates().contains(caseDetails.getState())
+                && DirectionType.APPEAL_TO_PROCEED.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())) {
+            System.out.println("HEREB");
+            caseData.setDateSentToDwp(LocalDate.now().toString());
+            caseData.setInterlocReviewState(AWAITING_ADMIN_ACTION.getId());
+
+        } else if (DirectionType.REFUSE_EXTENSION.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())
+            && ExtensionNextEvent.SEND_TO_LISTING.toString().equals(caseData.getExtensionNextEventDl().getValue().getCode())) {
+            System.out.println("HEREC");
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java:152]: remove println

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java [hunk 8]
```diff
@@ -124,11 +131,100 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
         return sscsCaseDataPreSubmitCallbackResponse;
     }
 
-    private void updateCaseAfterExtensionRefused(SscsCaseData caseData, String interlocReviewState, State state) {
+    @NotNull
+    private SscsCaseData updateCaseForDirectionType(CaseDetails<SscsCaseData> caseDetails, SscsCaseData caseData) {
+
+        System.out.println("HERE: " + caseData.getDirectionTypeDl().getValue().getCode());
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java:137]: remove println

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java [hunk 9]
```diff
@@ -124,11 +131,100 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
+            caseData = updateCaseAfterExtensionRefused(caseData, null, State.WITH_DWP);
+
+        } else if (DirectionTypeItemList.GRANT_REINSTATEMENT.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())
+            && reinstatementFeatureFlag) {
+            System.out.println("HERE GRANT");
+            caseData = updateCaseAfterReinstatementGranted(caseData);
+
+        } else if (DirectionTypeItemList.REFUSE_REINSTATEMENT.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())
+            && reinstatementFeatureFlag) {
+            System.out.println("HERE REFUSE");
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java:167]: remove println

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java [hunk 10]
```diff
@@ -124,11 +131,100 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
+            caseData = updateCaseAfterReinstatementRefused(caseData);
+
+        } else {
+            System.out.println("HERE DEFAULT");
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java:171]: remove println

File: src/IntegrationTests/resources/config/application_it.properties (hunk 1 above)
Comment (danhamilton) [reply src/IntegrationTests/resources/config/application_it.properties]: Added IT

**Review by ryandac — APPROVED** [review]

**Review by srochani — APPROVED** [review]

**Pull request merged**